frontend/src/components/bucket/BucketTable.vue

```diff
@@ -1,4 +1,5 @@
 <script setup lang="ts">
+import ProgressSpinner from 'primevue/progressspinner';
```

Comment:
i would create our own 'Loader' component in `/src/components/layout`. 
import it into AppLayout.vue .. so it wont need importing again in child components.
eg in datatables you can do `<template #loading><Loader /></template>`
and in `OidcLoginView.vue` you can replace `<ProgressSpinner />` with `<Loader />`. 
what do you think guys?

Reply:
also in Loader we can customise the look with something like this:
```css
.bcbox-loading { 
 background: white;
 opacity: .3;

 .p-progress-spinner { 
  position: absolute;
  top: 30%;
  left: 50%;
  transform: translate(-50%, -50%);
  opacity: 1; 
 }
}
```
see: https://primevue.org/progressspinner/#style

Reply:
I would say it's worth encapsulating into its own SFC as you recommend if and only if it has something we're embedding in as a part of the SFC definition like some default scss styling.
Also note that we should be re-routing the progress spinner through the @lib wrapper that we have instead of directly importing.